### DISCUSSION TITLE
How manully setup appsettings?

### DISCUSSION THREAD

**Discussion by YuriyMorozyuk95** [microsoft/hack-together-teams#10] [Q&A]
Unfortinetly on my company PC, I have issue with proxy and not able to Prepare Teams app dependencies, according not able to login to my Azure/Office365 accaunt from Visual Studio. For deployment I use FTPS to my Azure App Service.

![image](https://github.com/microsoft/hack-together-teams/assets/27745979/f5f936f0-1424-4969-8347-c6cbc3ea8a89)

As result I need setup everything manually. Or may be I can run bicep scripts some other way.

But any way I need to set properties in appsetting, to configure everything.
I have:
Azure App Serve - where I host bot.
Azure Bot with User Issigned Identity instead of Signle/MultiTenant - with configured connection to Client App Registration and teams chanel.
Client App registration - for reciaving token to call another Web Api secured by AAD other Server app registration. All permissions configured.

Could you advise from where I should populate values:

"AllowedHosts": "*",
"BOT_ID": "$botId$",
"BOT_PASSWORD": "$bot-password$",
"TeamsFx": {
		"Authentication": {
			"ClientId": "$clientId$",
			"ClientSecret": "$client-secret$",
			"OAuthAuthority": "$oauthAuthority$",
			"ApplicationIdUri": "$applicationIdUri$",
			"Bot": {
				"InitiateLoginEndpoint": "$initiateLoginEndpoint$"
			}
		}
	}

From code I understood that BOT_ID is MicrosoftAppId,
MicrosoftAppType is hardcoded in Program.cs I update it to UserAssignedMSI
BOT_PASSWORD I can skip according my MicrosoftAppType is UserAssignedMSI?
ClientId is Client App Registration Id?
ClientSecret is Client App Registration Secret Id?
From where should I populate OAuthAuthority?

ApplicationIdUri should be Application ID URI from Client App Registration? 
on which format it should be?
 - api://{Domain name}/botid-{Client App Registration Id} ?
 - api://botid-{Client App Registration Id} ?
 - api://{Client App Registration Id} ?

From where should I populate Bot::InitiateLoginEndpoint ?

Hope you will help me resolve my authorisation issues, thank you.

**Comment by therealjohn** [comment]
Have you reviewed the steps here? https://learn.microsoft.com/en-us/troubleshoot/developer/visualstudio/installation/troubleshoot-network-related-errors#error-proxy-authorization-required

You're correct, `BOT_ID` is the Client ID of the Azure AD App created for the Bot. 
The `BOT_PASSWORD` is the Client Secret for that same Azure AD App for the Bot.

Which project template are you using?

**Reply by YuriyMorozyuk95** [reply]
I am using C# Workflow bot template, also add autherisation code, by Porject > TeamsTolkit > Add Authorisation code.
![image](https://github.com/microsoft/hack-together-teams/assets/27745979/06e13813-2538-4656-889c-eb97778daff0)

If my bot type is MicrosoftAppType :  UserAssignedMSI, should I setup BOT_PASSWORD ?
And could you give advise regarding thouse properties?
![image](https://github.com/microsoft/hack-together-teams/assets/27745979/b8b3ed97-035c-49cf-8c24-2a3f28c552a8)

Here is how they are used:
![image](https://github.com/microsoft/hack-together-teams/assets/27745979/f575ce5f-987c-42ae-8a45-ed3bca0fd2c0)
